torch/fx/experimental/symbolic_shapes.py

```diff
@@ -5151,11 +5151,19 @@ def produce_guards_expression(
             return " and ".join(produced_guards)
         return None
 
-    def evaluate_symexpr(self, code: str) -> Union[int, float, bool]:
+    def evaluate_symexpr(
+        self, code: str, *, return_symints: bool
+    ) -> Union[int, float, bool]:
         """
         To be used by compile_fx to evaluate symexprs
         """
-        args = {str(e): val for e, val in self.var_to_val.items()}
+        if return_symints:
+            args = {
+                str(e): SymInt(SymNode(e, self, int, int(val), fx_node=None))
```

Comment:
I guess this is O(num_output_symints * num_total_symints) ..

Reply:
True. This seemed sus to I tried running a small benchmark (function with 100 3D inputs compiled with dynamic=True, P1681754466) with `TORCH_COMPILE_CPROFILE=1`, and:

(1) I don't see `evaluate_symexpr` or `set_tracing_context_output_strides` in the svg

(2) The E2E compile time was 15s - it seems like there is probably other low hanging fruit in a heavily-dynamic microbenchmark to find compile time wins

svg download command: `jf download GG0n2huLD3y3VGsBAAc3IKj-v5g0bsIXAAAz --file "_compile_inner_0_2.svg"`

Reply:
(15s was a lot, ended up filing an issue about it here 😛 https://github.com/pytorch/pytorch/issues/140970)